### PR TITLE
backup more namespaces in Embedded Cluster

### PR DESCRIPTION
**Pull request by laverya**
#### What this PR does / why we need it:

Embedded Cluster snapshots require resources outside of the `kotsadm` namespace.  This adds those namespaces to the backups created if it is an Embedded Cluster installation.

It also changes the label selector to use `replicated.app/disaster-recovery` labels instead, which application vendors will need to add to their application.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/kotsadmsnapshot/backup.go [hunk 1]
```diff
@@ -962,6 +958,11 @@ func prepareIncludedNamespaces(namespaces []string) []string {
 		uniqueNamespaces[n] = true
 	}
 
+	if util.IsEmbeddedCluster() {
+		uniqueNamespaces["embedded-cluster"] = true
+		uniqueNamespaces["kube-system"] = true
+	}
```
Comment (laverya) [inline pkg/kotsadmsnapshot/backup.go:964]: I think I should refactor this to require a boolean isEC argument

File: pkg/kotsadmsnapshot/backup.go [hunk 2]
```diff
@@ -962,6 +958,12 @@ func prepareIncludedNamespaces(namespaces []string) []string {
 		uniqueNamespaces[n] = true
 	}
 
+	if isEC {
+		uniqueNamespaces["embedded-cluster"] = true
+		uniqueNamespaces["kube-system"] = true
+		uniqueNamespaces["openebs"] = true
```
Comment (sgalsaleh) [inline pkg/kotsadmsnapshot/backup.go:964]: wouldn't `openebs` be installed along with k0s and velero in order to even initiate the restore process?

Comment (laverya) [reply pkg/kotsadmsnapshot/backup.go:964]: I think we install k0s + velero, then restore openebs, then restore kotsadm

Comment (sgalsaleh) [reply pkg/kotsadmsnapshot/backup.go:964]: how would velero be installed without openebs?

Comment (laverya) [reply pkg/kotsadmsnapshot/backup.go:965]: velero does not require in-cluster storage

Comment (sgalsaleh) [reply pkg/kotsadmsnapshot/backup.go:967]: 👍

File: pkg/kotsadmsnapshot/backup.go [hunk 3]
```diff
@@ -977,18 +982,34 @@ func excludeShutdownPodsFromBackup(ctx context.Context, clientset kubernetes.Int
 		"status.phase": string(corev1.PodFailed),
 	}
 
-	podListOption := metav1.ListOptions{
-		LabelSelector: labels.SelectorFromSet(veleroBackup.Spec.LabelSelector.MatchLabels).String(),
-		FieldSelector: fields.SelectorFromSet(selectorMap).String(),
+	labelSets := []string{}
+	if veleroBackup.Spec.LabelSelector.MatchLabels != nil {
```
Comment (sgalsaleh) [inline pkg/kotsadmsnapshot/backup.go:986]: Would this ever be `nil` given that it's initialized here? https://github.com/replicatedhq/kots/pull/4583/files#diff-5bf9dc41509334fa70fd5f20b3c75acac8ac8efa3b3ad54d1d18aa5ee2999792R1051

Comment (sgalsaleh) [reply pkg/kotsadmsnapshot/backup.go:986]: maybe use `len`?

Comment (laverya) [reply pkg/kotsadmsnapshot/backup.go:986]: oof, will do

Comment (laverya) [reply pkg/kotsadmsnapshot/backup.go]: done

**Review by cbodonnell — APPROVED** [review]

**Pull request merged**